File: assets/in.py
#!/usr/bin/env python3

import json
import sys

from util import calculate_checksum

if len(sys.argv) < 2:
    print("usage: {0} destination".format(sys.argv[0]))
    sys.exit(1)

destination = sys.argv[1]
payload = json.loads(sys.stdin.read())
source = payload.get("source", {})

filename = source.get("filename", "")

if not filename:
    print("source.filename is required")
    sys.exit(1)

content = source.get("content", "")

with open("{0}/{1}".format(destination, filename), "wb") as f:
    f.write(content)
    version = {
        "version": {"sha256sum": calculate_checksum(filename, content)}
    }
    print(json.dumps(version))

File: assets/in.js
#!/usr/bin/env node

const fs = require('fs')

const lib = require('./lib')

async function main () {
  try {
    const payload = await lib.populatePayloadAsync()
    return putFilesInWorkspace(payload)
  } catch (e) {
    console.error(e.message)
    return process.exit(1)
  }
}

function putFilesInWorkspace (input) {
  if (process.argv.length < 3) {
    console.error(`usage: ${process.argv[0]} destination`)
    return process.exit(1)
  }

  // NOTE node abc.js foobar
  // 0: node
  // 1: abc.js
  // 2: foobar
  const destination = process.argv[2]

  const { source = {} } = input
  const { filename, content } = source

  const digest = lib.calculateChecksum(filename, content)

  const stream = fs.createWriteStream(`${destination}/${filename}`)
  stream.write(content)

  console.log(JSON.stringify({ version: { sha256: digest }, metadata: [] }))
}

main()
